### PR TITLE
Split out id_minter_images and id_minter_works

### PR DESCRIPTION
**Pull request by warrd**

### CODE REVIEW

File: pipeline/id_minter/id_minter_images/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM wellcome/typesafe_config_base:104
+
+ADD target/universal/stage /opt/docker
+
+ENV PROJECT id_minter
```
Comment (alexwlchan) [inline pipeline/id_minter/id_minter_images/Dockerfile:5]:
```suggestion
ENV PROJECT id_minter_images
```

File: pipeline/id_minter/id_minter_images/src/main/scala/uk/ac/wellcome/platform/id_minter_images/Main.scala [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      )
+    )
+
+    new IdMinterWorkerService(
+      identifierGenerator = identifierGenerator,
+      sender = BigMessagingBuilder.buildBigMessageSender(config),
+      messageStream = BigMessagingBuilder.buildMessageStream[Json](config),
+      rdsClientConfig = RDSBuilder.buildRDSClientConfig(config),
+      identifiersTableConfig = identifiersTableConfig
+    )
```
Comment (alexwlchan) [inline pipeline/id_minter/id_minter_images/src/main/scala/uk/ac/wellcome/platform/id_minter_images/Main.scala:41]: Is anything about this images-specific, or could we push it into a common object/class?

File: pipeline/id_minter/id_minter_images/src/main/scala/uk/ac/wellcome/platform/id_minter_images/services/IdMinterWorkerService.scala [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  def processMessage(json: Json): Future[Unit] = Future fromTry {
+    for {
+      sourceIdentifiers <- SourceIdentifierEmbedder.scan(json)
+      mintedIdentifiers <- identifierGenerator.retrieveOrGenerateCanonicalIds(
+        sourceIdentifiers)
+      updatedJson <- SourceIdentifierEmbedder.update(json, mintedIdentifiers)
+      _ <- sender.sendT(updatedJson)
+    } yield ()
+  }
```
Comment (alexwlchan) [inline pipeline/id_minter/id_minter_images/src/main/scala/uk/ac/wellcome/platform/id_minter_images/services/IdMinterWorkerService.scala:53]: Ditto.

File: pipeline/id_minter/id_minter_works/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,5 @@
+FROM wellcome/typesafe_config_base:104
+
+ADD target/universal/stage /opt/docker
+
+ENV PROJECT id_minter
```
Comment (alexwlchan) [inline pipeline/id_minter/id_minter_works/Dockerfile:5]:
```suggestion
ENV PROJECT id_minter_works
```

**Review by alexwlchan — COMMENTED** [review]
I would be careful of letting the WorkerService and Main objects diverge, and put those in the common lib too, if we can.

**Comment by warrd** [comment]
At the moment they are the same, but the whole idea is that they need to diverge: the worker service for works will be using `pipeline_storage` whereas for images we are sticking to `big_messaging` for now.

**Review by alexwlchan — APPROVED** [review]
Duh, of course. Continue.

**Review by alexwlchan — APPROVED** [review]

**Pull request merged**